source/kra/parser.d

```diff
@@ -95,73 +95,30 @@ T getAttrValue(T, A...)(in A attributes, string name, T defaultValue) {
     return value.length > 1 ? to!T(value[1]) : defaultValue;
 }
 
-Layer baseLayer(T...)(in T attributes) {
-    Layer layer;
-    layer.name = getAttrValue!string(attributes, "name", "");
-    layer.isVisible = cast(bool) getAttrValue!int(attributes, "visible", 0);
-    layer.opacity = getAttrValue!int(attributes, "opacity", 255);
-    layer.uuid = getAttrValue!string(attributes, "uuid", "");
-    layer.x = getAttrValue!int(attributes, "x", 0);
-    layer.y = getAttrValue!int(attributes, "y", 0);
-    return layer;
-}
-
-void importAttributes(ref KRA kra, ref DOMEntity!string layerEntity) {
+void importAttributes(ref KRA kra, ref DOMEntity!string layerEntity, ref Layer[] layers) {
     auto layers = layerEntity.children.filter!(x => x.name == "layer");
 
     foreach (l; layers) {
         auto attrs = l.attributes;
 
+        auto fileName = getAttrValue!string(attrs, "filename", "");
+        enforce(fileName != "", "Invalid layer: filename attribute is required");
```

Comment:
A string may be null and not `""` in some instances. The correct way to do this check would be `fileName.length > 0`